dhall/tests/Dhall/Test/QuickCheck.hs

```diff
@@ -374,6 +374,15 @@ standardizedExpression (Prefer PreferFromCompletion _ _) =
     False
 standardizedExpression (Prefer (PreferFromWith _) _ _) =
     False
+-- The following three expressions are valid ASTs, but they can never be parsed,
+-- because the annotation will associate with `Merge`/`ListLit`/`ToMap` with
+-- higher precedence
```

Comment:
Thanks @Gabriel439!

I plan on adding parsers tests for the cases in https://github.com/dhall-lang/dhall-haskell/issues/1924#issuecomment-660679313, and I'll try to remember to update this comment to reference these tests.